File: suc/kotlin/010.kt
package com.example.singlefileui

import android.os.Bundle
import androidx.activity.ComponentActivity
import androidx.activity.compose.setContent
import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.lazy.LazyRow
import androidx.compose.foundation.lazy.items
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Button
import androidx.compose.material3.ButtonDefaults
import androidx.compose.material3.Card
import androidx.compose.material3.CardDefaults
import androidx.compose.material3.CenterAlignedTopAppBar
import androidx.compose.material3.ExperimentalMaterial3Api
import androidx.compose.material3.LinearProgressIndicator
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Scaffold
import androidx.compose.material3.Shapes
import androidx.compose.material3.Slider
import androidx.compose.material3.SliderDefaults
import androidx.compose.material3.Surface
import androidx.compose.material3.Text
import androidx.compose.material3.TopAppBarDefaults
import androidx.compose.material3.Typography
import androidx.compose.material3.lightColorScheme
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableFloatStateOf
import androidx.compose.runtime.mutableIntStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.setValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.TextStyle
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.tooling.preview.Preview
import androidx.compose.ui.unit.Dp
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import androidx.core.view.WindowCompat
import androidx.core.view.WindowInsetsCompat
import androidx.core.view.WindowInsetsControllerCompat

private const val NAME = "007_StoryEditor_en"
private const val UI_TYPE = "Social"
private const val STYLE_THEME = "Monochrome"
private const val LANG = "en"
private const val BASELINE_SIZE = "720x1280"

object AppTokens {
    object Colors {
        val primary = Color(0xFF1976D2)
        val secondary = Color(0xFF607D8B)
        val tertiary = Color(0xFF757575)
        val background = Color(0xFFECEFF1)
        val surface = Color(0xFFFFFFFF)
        val surfaceVariant = Color(0xFFCFD8DC)
        val outline = Color(0xFFB0BEC5)
        val success = Color(0xFF4CAF50)
        val warning = Color(0xFFFFC107)
        val error = Color(0xFFF44336)
        val onPrimary = Color(0xFFFFFFFF)
        val onSecondary = Color(0xFFFFFFFF)
        val onTertiary = Color(0xFFFFFFFF)
        val onBackground = Color(0xFF263238)
        val onSurface = Color(0xFF263238)
    }

    object TypographyTokens {
        val display = TextStyle(fontSize = 36.sp, fontWeight = FontWeight.Bold, letterSpacing = (-0.5).sp)
        val headline = TextStyle(fontSize = 24.sp, fontWeight = FontWeight.SemiBold, letterSpacing = 0.sp)
        val title = TextStyle(fontSize = 18.sp, fontWeight = FontWeight.Medium, letterSpacing = 0.1.sp)
        val body = TextStyle(fontSize = 14.sp, fontWeight = FontWeight.Normal, letterSpacing = 0.25.sp)
        val label = TextStyle(fontSize = 12.sp, fontWeight = FontWeight.Medium, letterSpacing = 0.5.sp)
    }

    object Shapes {
        val small = RoundedCornerShape(4.dp)
        val medium = RoundedCornerShape(8.dp)
        val large = RoundedCornerShape(12.dp)
    }

    object Spacing {
        val sm = 8.dp
        val md = 12.dp
        val lg = 16.dp
    }

    data class ShadowSpec(val elevation: Dp, val radius: Dp, val dy: Dp, val opacity: Float)

    object ElevationMapping {
        val level1 = ShadowSpec(1.dp, 3.dp, 1.dp, 0.1f)
        val level2 = ShadowSpec(3.dp, 6.dp, 2.dp, 0.1f)
    }
}

private val AppColorScheme = lightColorScheme(
    primary = AppTokens.Colors.primary,
    secondary = AppTokens.Colors.secondary,
    tertiary = AppTokens.Colors.tertiary,
    background = AppTokens.Colors.background,
    surface = AppTokens.Colors.surface,
    surfaceVariant = AppTokens.Colors.surfaceVariant,
    outline = AppTokens.Colors.outline,
    onPrimary = AppTokens.Colors.onPrimary,
    onSecondary = AppTokens.Colors.onSecondary,
    onTertiary = AppTokens.Colors.onTertiary,
    onBackground = AppTokens.Colors.onBackground,
    onSurface = AppTokens.Colors.onSurface,
    error = AppTokens.Colors.error
)

private val AppTypography = Typography(
    displayLarge = AppTokens.TypographyTokens.display,
    headlineMedium = AppTokens.TypographyTokens.headline,
    titleMedium = AppTokens.TypographyTokens.title,
    bodyMedium = AppTokens.TypographyTokens.body,
    labelMedium = AppTokens.TypographyTokens.label
)

@Composable
fun AppTheme(content: @Composable () -> Unit) {
    MaterialTheme(
        colorScheme = AppColorScheme,
        typography = AppTypography,
        shapes = Shapes(
            small = AppTokens.Shapes.small,
            medium = AppTokens.Shapes.medium,
            large = AppTokens.Shapes.large
        ),
        content = content
    )
}

data class StoryLayer(val id: Int, val type: String, val content: String)

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun RootScreen() {
    var textOpacity by remember { mutableFloatStateOf(1f) }
    var uploadProgress by remember { mutableFloatStateOf(0.65f) }
    val layers = remember {
        listOf(
            StoryLayer(1, "BG", "Gradient"),
            StoryLayer(2, "IMG", "User Photo"),
            StoryLayer(3, "TXT", "Hello World"),
            StoryLayer(4, "ICO", "Sticker")
        )
    }
    var selectedLayer by remember { mutableIntStateOf(layers[2].id) }

    Scaffold(
        contentWindowInsets = androidx.compose.foundation.layout.WindowInsets(0),
        topBar = {
            CenterAlignedTopAppBar(
                title = { Text("Story Editor", style = MaterialTheme.typography.titleMedium) },
                navigationIcon = {
                    Button(
                        onClick = {},
                        colors = ButtonDefaults.buttonColors(containerColor = Color.Transparent, contentColor = AppTokens.Colors.onBackground),
                        modifier = Modifier.padding(horizontal = AppTokens.Spacing.sm)
                    ) {
                        Text("Exit")
                    }
                },
                actions = {
                     Button(
                        onClick = {},
                        colors = ButtonDefaults.buttonColors(containerColor = AppTokens.Colors.primary, contentColor = AppTokens.Colors.onPrimary),
                         shape = AppTokens.Shapes.medium,
                         modifier = Modifier.padding(horizontal = AppTokens.Spacing.sm)
                    ) {
                        Text("Post")
                    }
                },
                colors = TopAppBarDefaults.centerAlignedTopAppBarColors(containerColor = AppTokens.Colors.surface)
            )
        },
        bottomBar = {
            Surface(
                color = AppTokens.Colors.surface,
                tonalElevation = AppTokens.ElevationMapping.level2.elevation,
                shadowElevation = AppTokens.ElevationMapping.level2.elevation
            ) {
                Column(modifier = Modifier.padding(AppTokens.Spacing.lg)) {
                    Text("Layer Properties", style = MaterialTheme.typography.labelMedium, color = AppTokens.Colors.secondary)
                    Spacer(modifier = Modifier.height(AppTokens.Spacing.md))
                    Row(verticalAlignment = Alignment.CenterVertically) {
                        Text("Opacity", style = MaterialTheme.typography.bodyMedium, modifier = Modifier.width(80.dp))
                        Slider(
                            value = textOpacity,
                            onValueChange = { textOpacity = it },
                            modifier = Modifier.weight(1f),
                            colors = SliderDefaults.colors(
                                thumbColor = AppTokens.Colors.primary,
                                activeTrackColor = AppTokens.Colors.primary,
                                inactiveTrackColor = AppTokens.Colors.surfaceVariant
                            )
                        )
                    }
                }
            }
        },
        containerColor = AppTokens.Colors.background
    ) { padding ->
        Column(
            modifier = Modifier
                .fillMaxSize()
                .padding(padding),
            horizontalAlignment = Alignment.CenterHorizontally,
            verticalArrangement = Arrangement.spacedBy(AppTokens.Spacing.lg)
        ) {
            Card(
                modifier = Modifier
                    .fillMaxWidth()
                    .weight(1f)
                    .padding(AppTokens.Spacing.lg),
                shape = AppTokens.Shapes.large,
                colors = CardDefaults.cardColors(containerColor = AppTokens.Colors.tertiary),
                elevation = CardDefaults.cardElevation(defaultElevation = AppTokens.ElevationMapping.level1.elevation)
            ) {
                Box(modifier = Modifier.fillMaxSize(), contentAlignment = Alignment.Center) {
                    Text(
                        text = "Your Story Preview",
                        style = MaterialTheme.typography.headlineMedium,
                        color = AppTokens.Colors.onTertiary.copy(alpha = textOpacity),
                        textAlign = TextAlign.Center
                    )
                }
            }

            Column(
                modifier = Modifier
                    .fillMaxWidth()
                    .padding(horizontal = AppTokens.Spacing.lg),
                verticalArrangement = Arrangement.spacedBy(AppTokens.Spacing.sm)
            ) {
                Text( "Upload Status", style = MaterialTheme.typography.labelMedium, color = AppTokens.Colors.secondary)
                LinearProgressIndicator(
                    progress = { uploadProgress },
                    modifier = Modifier.fillMaxWidth().height(AppTokens.Spacing.sm),
                    color = AppTokens.Colors.success,
                    trackColor = AppTokens.Colors.surfaceVariant
                )
            }

            LazyRow(
                horizontalArrangement = Arrangement.spacedBy(AppTokens.Spacing.md),
                modifier = Modifier.fillMaxWidth(),
                contentPadding = androidx.compose.foundation.layout.PaddingValues(horizontal = AppTokens.Spacing.lg, vertical = AppTokens.Spacing.md)
            ) {
                items(layers) { layer ->
                    val isSelected = layer.id == selectedLayer
                    Card(
                        onClick = { selectedLayer = layer.id },
                        shape = AppTokens.Shapes.medium,
                        colors = CardDefaults.cardColors(containerColor = if(isSelected) AppTokens.Colors.primary else AppTokens.Colors.surface),
                        modifier = Modifier.border(
                            width = 1.dp,
                            color = if(isSelected) AppTokens.Colors.primary else AppTokens.Colors.outline,
                            shape = AppTokens.Shapes.medium
                        )
                    ) {
                        Column(
                            modifier = Modifier.padding(horizontal = AppTokens.Spacing.lg, vertical = AppTokens.Spacing.md).width(100.dp),
                            horizontalAlignment = Alignment.CenterHorizontally,
                            verticalArrangement = Arrangement.spacedBy(AppTokens.Spacing.sm)
                        ) {
                             Box(
                                modifier = Modifier.size(32.dp).background(if(isSelected) AppTokens.Colors.onPrimary.copy(alpha=0.2f) else AppTokens.Colors.surfaceVariant, CircleShape)
                            )
                            Text(layer.type, style = MaterialTheme.typography.labelMedium, color = if(isSelected) AppTokens.Colors.onPrimary else AppTokens.Colors.onSurface.copy(alpha = 0.7f))
                            Text(layer.content, style = MaterialTheme.typography.bodyMedium, color = if(isSelected) AppTokens.Colors.onPrimary else AppTokens.Colors.onSurface)
                        }
                    }
                }
            }
        }
    }
}

class MainActivity : ComponentActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        WindowCompat.setDecorFitsSystemWindows(window, false)
        val controller = WindowInsetsControllerCompat(window, window.decorView)
        controller.systemBarsBehavior = WindowInsetsControllerCompat.BEHAVIOR_SHOW_TRANSIENT_BARS_BY_SWIPE
        controller.hide(WindowInsetsCompat.Type.systemBars())
        setContent {
            AppTheme {
                Surface(color = MaterialTheme.colorScheme.background) {
                    RootScreen()
                }
            }
        }
    }

    override fun onWindowFocusChanged(hasFocus: Boolean) {
        super.onWindowFocusChanged(hasFocus)
        if (hasFocus) {
            val controller = WindowInsetsControllerCompat(window, window.decorView)
            controller.systemBarsBehavior = WindowInsetsControllerCompat.BEHAVIOR_SHOW_TRANSIENT_BARS_BY_SWIPE
            controller.hide(WindowInsetsCompat.Type.systemBars())
        }
    }
}

@Preview(showBackground = true, backgroundColor = 0xFFECEFF1)
@Composable
fun PreviewRoot() {
    AppTheme {
        RootScreen()
    }
}

File: suc/swift/010.swift
import SwiftUI

// MARK: - AppTokens
struct AppTokens {
    struct Colors {
        static let primary = Color(hex: 0xFF1976D2)
        static let secondary = Color(hex: 0xFF607D8B)
        static let tertiary = Color(hex: 0xFF757575)
        static let background = Color(hex: 0xFFECEFF1)
        static let surface = Color(hex: 0xFFFFFFFF)
        static let surfaceVariant = Color(hex: 0xFFCFD8DC)
        static let outline = Color(hex: 0xFFB0BEC5)
        static let success = Color(hex: 0xFF4CAF50)
        static let warning = Color(hex: 0xFFFFC107)
        static let error = Color(hex: 0xFFF44336)
        static let onPrimary = Color(hex: 0xFFFFFFFF)
        static let onSecondary = Color(hex: 0xFFFFFFFF)
        static let onTertiary = Color(hex: 0xFFFFFFFF)
        static let onBackground = Color(hex: 0xFF263238)
        static let onSurface = Color(hex: 0xFF263238)
    }

    struct TypographyTokens {
        static let display = Font.system(size: 36, weight: .bold)
        static let headline = Font.system(size: 24, weight: .semibold)
        static let title = Font.system(size: 18, weight: .medium)
        static let body = Font.system(size: 14, weight: .regular)
        static let label = Font.system(size: 12, weight: .medium)
    }

    struct Shapes {
        static let small = RoundedRectangle(cornerRadius: 4)
        static let medium = RoundedRectangle(cornerRadius: 8)
        static let large = RoundedRectangle(cornerRadius: 12)
    }

    struct Spacing {
        static let sm: CGFloat = 8
        static let md: CGFloat = 12
        static let lg: CGFloat = 16
        static let xl: CGFloat = 24 // 增加到24
        static let xxl: CGFloat = 32 // 新增更大的间距
    }
    
    static let topBarHeight: CGFloat = 64
}

// 扩展 Color 以支持从十六进制值初始化
extension Color {
    init(hex: UInt) {
        let alpha = Double((hex >> 24) & 0xFF) / 255.0
        let red = Double((hex >> 16) & 0xFF) / 255.0
        let green = Double((hex >> 8) & 0xFF) / 255.0
        let blue = Double(hex & 0xFF) / 255.0
        
        self.init(
            .sRGB,
            red: red,
            green: green,
            blue: blue,
            opacity: alpha
        )
    }
}

// MARK: - Data Models
struct StoryLayer: Identifiable {
    let id: Int
    let type: String
    let content: String
}

// MARK: - RootScreen
struct RootScreen: View {
    @State private var textOpacity: Double = 1.0
    @State private var uploadProgress: Double = 0.65
    @State private var layers: [StoryLayer] = [
        StoryLayer(id: 1, type: "BG", content: "Gradient"),
        StoryLayer(id: 2, type: "IMG", content: "User Photo"),
        StoryLayer(id: 3, type: "TXT", content: "Hello World"),
        StoryLayer(id: 4, type: "ICO", content: "Sticker")
    ]
    @State private var selectedLayer: Int = 3

    var body: some View {
        ZStack(alignment: .bottom) {
            // 主内容区域
            VStack(spacing: 0) {
                // 顶部栏 - 对应 CenterAlignedTopAppBar
                HStack {
                    // 左侧按钮 - Exit (透明背景)
                    Button(action: {}) {
                        Text("Exit")
                            .font(AppTokens.TypographyTokens.body)
                            .foregroundColor(AppTokens.Colors.onBackground)
                    }
                    .padding(.horizontal, AppTokens.Spacing.sm)
                    
                    Spacer()
                    
                    // 中间标题
                    Text("Story Editor")
                        .font(AppTokens.TypographyTokens.title)
                        .foregroundColor(AppTokens.Colors.onSurface)
                    
                    Spacer()
                    
                    // 右侧按钮 - Post (蓝色背景)
                    Button(action: {}) {
                        Text("Post")
                            .font(AppTokens.TypographyTokens.body)
                            .foregroundColor(AppTokens.Colors.onPrimary)
                            .padding(.horizontal, AppTokens.Spacing.md)
                            .padding(.vertical, AppTokens.Spacing.sm)
                    }
                    .background(AppTokens.Colors.primary)
                    .clipShape(AppTokens.Shapes.medium)
                    .padding(.horizontal, AppTokens.Spacing.sm)
                }
                .frame(height: AppTokens.topBarHeight)
                .background(AppTokens.Colors.surface)
                
                // 主要内容区域 - 对应 Scaffold 的 content
                ScrollView {
                    VStack(spacing: AppTokens.Spacing.lg) {
                        // 预览卡片 - 对应 Card with weight(1f)
                        ZStack {
                            AppTokens.Shapes.large
                                .fill(AppTokens.Colors.tertiary)
                                .shadow(color: Color.black.opacity(0.1), radius: 3, x: 0, y: 1)
                            
                            Text("Your Story Preview")
                                .font(AppTokens.TypographyTokens.headline)
                                .foregroundColor(AppTokens.Colors.onTertiary.opacity(textOpacity))
                                .multilineTextAlignment(.center)
                        }
                        .frame(height: 200)
                        .padding(.horizontal, AppTokens.Spacing.lg)
                        
                        // 上传状态 - 对应 Column
                        VStack(alignment: .leading, spacing: AppTokens.Spacing.sm) {
                            Text("Upload Status")
                                .font(AppTokens.TypographyTokens.label)
                                .foregroundColor(AppTokens.Colors.secondary)
                            
                            // 自定义进度条 - 对应 LinearProgressIndicator
                            GeometryReader { geometry in
                                ZStack(alignment: .leading) {
                                    // 轨道背景
                                    RoundedRectangle(cornerRadius: 4)
                                        .fill(AppTokens.Colors.surfaceVariant)
                                        .frame(height: AppTokens.Spacing.sm)
                                    
                                    // 进度条
                                    RoundedRectangle(cornerRadius: 4)
                                        .fill(AppTokens.Colors.success)
                                        .frame(width: geometry.size.width * CGFloat(uploadProgress), height: AppTokens.Spacing.sm)
                                }
                            }
                            .frame(height: AppTokens.Spacing.sm)
                        }
                        .padding(.horizontal, AppTokens.Spacing.lg)
                        
                        // 图层列表 - 对应 LazyRow
                        VStack(alignment: .leading, spacing: AppTokens.Spacing.sm) {
                            Text("Story Layers")
                                .font(AppTokens.TypographyTokens.label)
                                .foregroundColor(AppTokens.Colors.secondary)
                                .padding(.horizontal, AppTokens.Spacing.lg)
                            
                            ScrollView(.horizontal, showsIndicators: false) {
                                HStack(spacing: AppTokens.Spacing.lg) {
                                    ForEach(layers) { layer in
                                        LayerItemView(
                                            layer: layer,
                                            isSelected: layer.id == selectedLayer,
                                            onSelect: { selectedLayer = layer.id }
                                        )
                                    }
                                }
                                .padding(.horizontal, AppTokens.Spacing.lg)
                                .padding(.vertical, AppTokens.Spacing.md)
                            }
                        }
                    }
                    .padding(.vertical, AppTokens.Spacing.lg)
                }
                .background(AppTokens.Colors.background)
            }
            
            // 底部栏 - 对应 BottomBar 的 Surface
            VStack(alignment: .leading, spacing: AppTokens.Spacing.md) {
                Text("Layer Properties")
                    .font(AppTokens.TypographyTokens.label)
                    .foregroundColor(AppTokens.Colors.secondary)
                
                HStack {
                    Text("Opacity")
                        .font(AppTokens.TypographyTokens.body)
                        .frame(width: 80, alignment: .leading)
                    
                    Slider(value: $textOpacity, in: 0...1)
                        .tint(AppTokens.Colors.primary)
                }
            }
            .padding(AppTokens.Spacing.lg)
            .frame(maxWidth: .infinity)
            .background(AppTokens.Colors.surface)
            .shadow(color: Color.black.opacity(0.1), radius: 6, x: 0, y: -2)
        }
        .background(AppTokens.Colors.background)
        .ignoresSafeArea(.all, edges: .all)
        .statusBar(hidden: true) // 隐藏状态栏
    }
}

// MARK: - Layer Item View
struct LayerItemView: View {
    let layer: StoryLayer
    let isSelected: Bool
    let onSelect: () -> Void
    
    var body: some View {
        Button(action: onSelect) {
            VStack(spacing: AppTokens.Spacing.sm) {
                // 图标区域 - 对应 Box with CircleShape
                Circle()
                    .fill(isSelected ? AppTokens.Colors.onPrimary.opacity(0.2) : AppTokens.Colors.surfaceVariant)
                    .frame(width: 32, height: 32)
                
                // 类型标签
                Text(layer.type)
                    .font(AppTokens.TypographyTokens.label)
                    .foregroundColor(isSelected ? AppTokens.Colors.onPrimary : AppTokens.Colors.onSurface.opacity(0.7))
                
                // 内容文本
                Text(layer.content)
                    .font(AppTokens.TypographyTokens.body)
                    .foregroundColor(isSelected ? AppTokens.Colors.onPrimary : AppTokens.Colors.onSurface)
                    .multilineTextAlignment(.center)
                    .lineLimit(2)
            }
            .padding(.horizontal, AppTokens.Spacing.xxl) // 使用更大的水平内边距
            .padding(.vertical, AppTokens.Spacing.md)
            .frame(width: 130) // 进一步增加宽度以容纳更大的内边距
            .background(isSelected ? AppTokens.Colors.primary : AppTokens.Colors.surface)
            .cornerRadius(8)
            .overlay(
                RoundedRectangle(cornerRadius: 8)
                    .stroke(isSelected ? AppTokens.Colors.primary : AppTokens.Colors.outline, lineWidth: 1)
            )
        }
        .buttonStyle(PlainButtonStyle())
    }
}

// MARK: - App Entry Point
@main
struct StoryEditorApp: App {
    var body: some Scene {
        WindowGroup {
            RootScreen()
        }
    }
}

// MARK: - Preview Provider
struct RootScreen_Previews: PreviewProvider {
    static var previews: some View {
        RootScreen()
    }
}